File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/SessionChatHistorySummary.java
package com.oracle.aidataplatform.dp.model;



/**
 * A single message in a chat session history. Does not include internal state.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=SessionChatHistorySummary.Builder.class)

public final class SessionChatHistorySummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"key", "sessionKey", "role", "timeCreated", "content", "toolName", "toolCallId", "metadata"})
    public SessionChatHistorySummary(String key, String sessionKey, String role, java.util.Date timeCreated, java.util.List<ChatMessage> content, String toolName, String toolCallId, java.util.Map<String, Object> metadata) {
        super();
        this.key = key;
        this.sessionKey = sessionKey;
        this.role = role;
        this.timeCreated = timeCreated;
        this.content = content;
        this.toolName = toolName;
        this.toolCallId = toolCallId;
        this.metadata = metadata;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Unique identifier for the message.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * Unique identifier for the message.
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * Identifier of the chat session this message belongs to.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sessionKey")
private String sessionKey;

        /**
         * Identifier of the chat session this message belongs to.
         * @param sessionKey the value to set
         * @return this builder
         **/
        

public Builder sessionKey(String sessionKey) {
    this.sessionKey = sessionKey;
    return this;
}
            /**
     * Role associated with the message, such as user, assistant, system, or tool.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("role")
private String role;

        /**
         * Role associated with the message, such as user, assistant, system, or tool.
         * @param role the value to set
         * @return this builder
         **/
        

public Builder role(String role) {
    this.role = role;
    return this;
}
            /**
     * Time the message was created.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

        /**
         * Time the message was created.
         * @param timeCreated the value to set
         * @return this builder
         **/
        

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
            /**
     * Message content.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("content")
private java.util.List<ChatMessage> content;

        /**
         * Message content.
         * @param content the value to set
         * @return this builder
         **/
        

public Builder content(java.util.List<ChatMessage> content) {
    this.content = content;
    return this;
}
            /**
     * Name of the tool, if this is a tool message.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("toolName")
private String toolName;

        /**
         * Name of the tool, if this is a tool message.
         * @param toolName the value to set
         * @return this builder
         **/
        

public Builder toolName(String toolName) {
    this.toolName = toolName;
    return this;
}
            /**
     * Identifier of the tool call, if applicable.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("toolCallId")
private String toolCallId;

        /**
         * Identifier of the tool call, if applicable.
         * @param toolCallId the value to set
         * @return this builder
         **/
        

public Builder toolCallId(String toolCallId) {
    this.toolCallId = toolCallId;
    return this;
}
            /**
     * Optional key-value metadata associated with the message
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("metadata")
private java.util.Map<String, Object> metadata;

        /**
         * Optional key-value metadata associated with the message
         * @param metadata the value to set
         * @return this builder
         **/
        

public Builder metadata(java.util.Map<String, Object> metadata) {
    this.metadata = metadata;
    return this;
}


        public SessionChatHistorySummary build() {
            SessionChatHistorySummary model = new SessionChatHistorySummary(this.key
                , this.sessionKey
                , this.role
                , this.timeCreated
                , this.content
                , this.toolName
                , this.toolCallId
                , this.metadata);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(SessionChatHistorySummary model) {
                this.key(model.getKey());
    this.sessionKey(model.getSessionKey());
    this.role(model.getRole());
    this.timeCreated(model.getTimeCreated());
    this.content(model.getContent());
    this.toolName(model.getToolName());
    this.toolCallId(model.getToolCallId());
    this.metadata(model.getMetadata());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Unique identifier for the message.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * Unique identifier for the message.
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * Identifier of the chat session this message belongs to.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sessionKey")
    private final String sessionKey;

        /**
     * Identifier of the chat session this message belongs to.
     * @return the value
     **/
    
    public String getSessionKey() {
        return sessionKey;
    }


        /**
     * Role associated with the message, such as user, assistant, system, or tool.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("role")
    private final String role;

        /**
     * Role associated with the message, such as user, assistant, system, or tool.
     * @return the value
     **/
    
    public String getRole() {
        return role;
    }


        /**
     * Time the message was created.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * Time the message was created.
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * Message content.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("content")
    private final java.util.List<ChatMessage> content;

        /**
     * Message content.
     * @return the value
     **/
    
    public java.util.List<ChatMessage> getContent() {
        return content;
    }


        /**
     * Name of the tool, if this is a tool message.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toolName")
    private final String toolName;

        /**
     * Name of the tool, if this is a tool message.
     * @return the value
     **/
    
    public String getToolName() {
        return toolName;
    }


        /**
     * Identifier of the tool call, if applicable.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toolCallId")
    private final String toolCallId;

        /**
     * Identifier of the tool call, if applicable.
     * @return the value
     **/
    
    public String getToolCallId() {
        return toolCallId;
    }


        /**
     * Optional key-value metadata associated with the message
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("metadata")
    private final java.util.Map<String, Object> metadata;

        /**
     * Optional key-value metadata associated with the message
     * @return the value
     **/
    
    public java.util.Map<String, Object> getMetadata() {
        return metadata;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("SessionChatHistorySummary(");
        sb.append("key=").append(String.valueOf(this.key));
        sb.append(", sessionKey=").append(String.valueOf(this.sessionKey));
        sb.append(", role=").append(String.valueOf(this.role));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", content=").append(String.valueOf(this.content));
        sb.append(", toolName=").append(String.valueOf(this.toolName));
        sb.append(", toolCallId=").append(String.valueOf(this.toolCallId));
        sb.append(", metadata=").append(String.valueOf(this.metadata));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof SessionChatHistorySummary)) {
            return false;
        }

        SessionChatHistorySummary other = (SessionChatHistorySummary) o;
        return java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.sessionKey, other.sessionKey) &&
            java.util.Objects.equals(this.role, other.role) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.content, other.content) &&
            java.util.Objects.equals(this.toolName, other.toolName) &&
            java.util.Objects.equals(this.toolCallId, other.toolCallId) &&
            java.util.Objects.equals(this.metadata, other.metadata);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.sessionKey == null ? 43 : this.sessionKey.hashCode());
        result = (result * PRIME) + (this.role == null ? 43 : this.role.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.content == null ? 43 : this.content.hashCode());
        result = (result * PRIME) + (this.toolName == null ? 43 : this.toolName.hashCode());
        result = (result * PRIME) + (this.toolCallId == null ? 43 : this.toolCallId.hashCode());
        result = (result * PRIME) + (this.metadata == null ? 43 : this.metadata.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/session_chat_history_summary.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class SessionChatHistorySummary(object):
    """
    A single message in a chat session history. Does not include internal state.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new SessionChatHistorySummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this SessionChatHistorySummary.
        :type key: str

        :param session_key:
            The value to assign to the session_key property of this SessionChatHistorySummary.
        :type session_key: str

        :param role:
            The value to assign to the role property of this SessionChatHistorySummary.
        :type role: str

        :param time_created:
            The value to assign to the time_created property of this SessionChatHistorySummary.
        :type time_created: datetime

        :param content:
            The value to assign to the content property of this SessionChatHistorySummary.
        :type content: list[oci.aidataplatform_dp.models.ChatMessage]

        :param tool_name:
            The value to assign to the tool_name property of this SessionChatHistorySummary.
        :type tool_name: str

        :param tool_call_id:
            The value to assign to the tool_call_id property of this SessionChatHistorySummary.
        :type tool_call_id: str

        :param metadata:
            The value to assign to the metadata property of this SessionChatHistorySummary.
        :type metadata: dict(str, object)

        """
        self.swagger_types = {
            'key': 'str',
            'session_key': 'str',
            'role': 'str',
            'time_created': 'datetime',
            'content': 'list[ChatMessage]',
            'tool_name': 'str',
            'tool_call_id': 'str',
            'metadata': 'dict(str, object)'
        }

        self.attribute_map = {
            'key': 'key',
            'session_key': 'sessionKey',
            'role': 'role',
            'time_created': 'timeCreated',
            'content': 'content',
            'tool_name': 'toolName',
            'tool_call_id': 'toolCallId',
            'metadata': 'metadata'
        }

        self._key = None
        self._session_key = None
        self._role = None
        self._time_created = None
        self._content = None
        self._tool_name = None
        self._tool_call_id = None
        self._metadata = None

    @property
    def key(self):
        """
        **[Required]** Gets the key of this SessionChatHistorySummary.
        Unique identifier for the message.


        :return: The key of this SessionChatHistorySummary.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this SessionChatHistorySummary.
        Unique identifier for the message.


        :param key: The key of this SessionChatHistorySummary.
        :type: str
        """
        self._key = key

    @property
    def session_key(self):
        """
        **[Required]** Gets the session_key of this SessionChatHistorySummary.
        Identifier of the chat session this message belongs to.


        :return: The session_key of this SessionChatHistorySummary.
        :rtype: str
        """
        return self._session_key

    @session_key.setter
    def session_key(self, session_key):
        """
        Sets the session_key of this SessionChatHistorySummary.
        Identifier of the chat session this message belongs to.


        :param session_key: The session_key of this SessionChatHistorySummary.
        :type: str
        """
        self._session_key = session_key

    @property
    def role(self):
        """
        **[Required]** Gets the role of this SessionChatHistorySummary.
        Role associated with the message, such as user, assistant, system, or tool.


        :return: The role of this SessionChatHistorySummary.
        :rtype: str
        """
        return self._role

    @role.setter
    def role(self, role):
        """
        Sets the role of this SessionChatHistorySummary.
        Role associated with the message, such as user, assistant, system, or tool.


        :param role: The role of this SessionChatHistorySummary.
        :type: str
        """
        self._role = role

    @property
    def time_created(self):
        """
        **[Required]** Gets the time_created of this SessionChatHistorySummary.
        Time the message was created.


        :return: The time_created of this SessionChatHistorySummary.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this SessionChatHistorySummary.
        Time the message was created.


        :param time_created: The time_created of this SessionChatHistorySummary.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def content(self):
        """
        **[Required]** Gets the content of this SessionChatHistorySummary.
        Message content.


        :return: The content of this SessionChatHistorySummary.
        :rtype: list[oci.aidataplatform_dp.models.ChatMessage]
        """
        return self._content

    @content.setter
    def content(self, content):
        """
        Sets the content of this SessionChatHistorySummary.
        Message content.


        :param content: The content of this SessionChatHistorySummary.
        :type: list[oci.aidataplatform_dp.models.ChatMessage]
        """
        self._content = content

    @property
    def tool_name(self):
        """
        Gets the tool_name of this SessionChatHistorySummary.
        Name of the tool, if this is a tool message.


        :return: The tool_name of this SessionChatHistorySummary.
        :rtype: str
        """
        return self._tool_name

    @tool_name.setter
    def tool_name(self, tool_name):
        """
        Sets the tool_name of this SessionChatHistorySummary.
        Name of the tool, if this is a tool message.


        :param tool_name: The tool_name of this SessionChatHistorySummary.
        :type: str
        """
        self._tool_name = tool_name

    @property
    def tool_call_id(self):
        """
        Gets the tool_call_id of this SessionChatHistorySummary.
        Identifier of the tool call, if applicable.


        :return: The tool_call_id of this SessionChatHistorySummary.
        :rtype: str
        """
        return self._tool_call_id

    @tool_call_id.setter
    def tool_call_id(self, tool_call_id):
        """
        Sets the tool_call_id of this SessionChatHistorySummary.
        Identifier of the tool call, if applicable.


        :param tool_call_id: The tool_call_id of this SessionChatHistorySummary.
        :type: str
        """
        self._tool_call_id = tool_call_id

    @property
    def metadata(self):
        """
        Gets the metadata of this SessionChatHistorySummary.
        Optional key-value metadata associated with the message


        :return: The metadata of this SessionChatHistorySummary.
        :rtype: dict(str, object)
        """
        return self._metadata

    @metadata.setter
    def metadata(self, metadata):
        """
        Sets the metadata of this SessionChatHistorySummary.
        Optional key-value metadata associated with the message


        :param metadata: The metadata of this SessionChatHistorySummary.
        :type: dict(str, object)
        """
        self._metadata = metadata

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
